pkg/apis/pipeline/v1/taskref_types_test.go

```diff
@@ -0,0 +1,54 @@
+package v1
+
+import "testing"
+
+func TestTaskRef_IsCustomTask(t *testing.T) {
+	tests := []struct {
+		name string
+		tr   *TaskRef
+		want bool
+	}{{
+		name: "not a custom task - apiVersion and Kind are not set",
+		tr: &TaskRef{
+			Name: "foo",
+		},
+		want: false,
+	}, {
+		name: "not a custom task - apiVersion is not set",
+		tr: &TaskRef{
+			Name: "foo",
+			Kind: "Example",
```

Comment:
I don't think it should be a `Custom Task` because there's no `APIVersion` - so the current behavior of `IsCustomTask` is correct in my opinion

What happens today is that this is considered a `Task`, which is not right and should be bug which is out of scope of this change - will open a separate issue to track it

Reply:
opened https://github.com/tektoncd/pipeline/issues/6459

Reply:
Thanks! Could you add a comment linking to that issue?